src/OAuthClient/ItemClient.php

```diff
@@ -1,42 +1,78 @@
 use NYPL\HoldRequestResultConsumer\Model\DataModel\Item;
-use NYPL\Starter\APIException;
+use NYPL\HoldRequestResultConsumer\Model\Exception\NonRetryableException;
+use NYPL\HoldRequestResultConsumer\Model\Exception\RetryableException;
 use NYPL\Starter\APILogger;
 use NYPL\Starter\Config;
+use NYPL\Starter\Model\Response\ErrorResponse;
 
 class ItemClient extends APIClient
 {
     /**
      * @param string $itemId
      * @param $nyplSource
-     * @return Item | null
+     * @return null|Item
+     * @throws NonRetryableException
+     * @throws RetryableException
      */
     public static function getItemByIdAndSource($itemId = '', $nyplSource)
     {
         $url = Config::get('API_ITEM_URL') . '/' . $nyplSource . '/' . $itemId;
 
         APILogger::addDebug('Retrieving item by Id and Source', (array) $url);
 
-        $response = self::get($url);
+        try {
+            $response = self::get($url);
 
-        $response = json_decode((string) $response->getBody(), true);
+            $statusCode = $response->getStatusCode();
 
-        APILogger::addDebug(
-            'Retrieve item by id and source',
-            $response['data']
-        );
+            $response = json_decode((string)$response->getBody(), true);
 
-        if ($response['statusCode'] !== 200) {
-            APILogger::addError(
-                'Failed',
-                array('Failed to retrieve item ', $itemId, $response['type'], $response['message'])
+            APILogger::addDebug(
+                'Retrieve item by id and source',
+                $response['data']
             );
-            return null;
-        }
 
-        return new Item($response['data']);
+            // Check statusCode range
+            if ($statusCode === 200) {
+                return new Item($response['data']);
+            } else {
+                APILogger::addError(
+                    'Failed',
+                    array('Failed to retrieve item ', $itemId, $response['type'], $response['message'])
+                );
+                return null;
+            }
+        } catch (ServerException $exception) {
+            throw new RetryableException(
+                'Server Error from ' . __FUNCTION__ . $exception->getMessage(),
+                'Server Error from ' . __FUNCTION__ . $exception->getMessage(),
+                $exception->getResponse()->getStatusCode(),
+                null,
+                $exception->getResponse()->getStatusCode(),
+                new ErrorResponse(
+                    $exception->getResponse()->getStatusCode(),
+                    'internal-server-error',
+                    'Server Error from ' . __FUNCTION__ . $exception->getMessage()
+                )
+            );
+        } catch (ClientException $exception) {
+            throw new NonRetryableException(
+                'Client Error from '. __FUNCTION__ . $exception->getMessage(),
+                'Client Error from '. __FUNCTION__ . $exception->getMessage(),
```

Comment:
Same as above for exception params; second param can be null instead of doubling the message.